scrapinghub/utils.py

```diff
@@ -112,23 +113,53 @@ def format_iter_filters(params):
 def parse_auth(auth):
     """Parse authentification token.
 
+    >>> parse_auth(None)  # noqa
+    RuntimeError: No API key provided and SH_APIKEY environment variable not set
+    >>> os.environ['SH_APIKEY'] = 'apikey'
     >>> parse_auth(None)
+    ('apikey', '')
     >>> parse_auth(('user', 'pass'))
     ('user', 'pass')
     >>> parse_auth('user:pass')
     ('user', 'pass')
-    >>> parse_auth('apikey')
-    ('apikey', '')
+    >>> parse_auth('c3a3c298c2b8c3a6c291c284c3a9')
+    ('c3a3c298c2b8c3a6c291c284c3a9', '')
     >>> parse_auth('312f322f333a736f6d652e6a77742e746f6b656e')
     ('1/2/3', 'some.jwt.token')
     """
-    if auth is None or isinstance(auth, tuple):
+    if auth is None:
+        apikey = os.environ.get('SH_APIKEY')
+        if apikey is None:
+            raise RuntimeError("No API key provided and SH_APIKEY "
+                               "environment variable not set")
+        return (apikey, '')
+
+    if isinstance(auth, tuple):
+        if len(auth) != 2:
+            raise ValueError("Wrong amount of authentication credentials")
+        if not isinstance(auth[0], string_types):
+            raise ValueError("Login must me of a string type")
+        if not (auth[1] is None or isinstance(auth[1], string_types)):
+            raise ValueError("Password must be None or of a string type")
```

Comment:
let's keep it simple - both loging and password can be strings only, if password isn't set it's empty string